synapse/util/stringutils.py

```diff
@@ -123,6 +126,8 @@ def parse_and_validate_server_name(server_name: str) -> Tuple[str, Optional[int]
     # that nobody is sneaking IP literals in that look like hostnames, etc.
 
     # look for ipv6 literals
+    if not host:
+        raise ValueError(f"Server name '{server_name}' has an invalid format.")
```

Comment:
minor: I'd usually go for a format string here, but prefer consistency with the other code in this method:
```suggestion
        raise ValueError("Server name '%s' has an invalid format" % (server_name,))
```